src/js/render_wordle.js

```diff
@@ -78,8 +78,80 @@ inverted_wordles.drawLayout = function (layout, words, instance) {
         .text(function (d) { return d.text; });
 };
 
+/** Return the computed value of the font size of the given DOM element
+ * @param {Object} elm - A DOM element
+ * @return {Number} The computed value of the font size of the given element
+ */
+inverted_wordles.getFontSizeValue = function (elm) {
+    const style = window.getComputedStyle(elm, null).getPropertyValue("font-size");
+    return parseFloat(style);
+};
+
+/** Find font sizes of all text nodes and sort unique values into an array. The array is returned by attaching
+ * to the wordle instance
+ * @param {WordleInstance} instance - The singleton instance
+ * @param {Object[]} textElements - An array of DOM elements for wordle texts
+ */
+inverted_wordles.getSortedUniqueFontSizes = function (instance, textElements) {
+    let fontSizes = [];
+    textElements.forEach(elm => {
+        fontSizes.push(inverted_wordles.getFontSizeValue(elm));
+    });
+
+    const uniqueFontSizes = fontSizes.filter((value, index, selfArray) => selfArray.indexOf(value) === index);
+    instance.sortedUniqueFontSizes = [...uniqueFontSizes].sort((a, b) => a - b);
+};
+
+/** Calculate the pitch value based on the font size
+ * @param {Number} thisFontSize - The font size The font size that is based upon to calculate the pitch value
+ * @param {Number[]} sortedUniqueFontSizes - Sorted and unique values of all font sizes
+ * @return {Number} The calculated pitch value in two decimal points
+ */
+inverted_wordles.calculatePitch = function (thisFontSize, sortedUniqueFontSizes) {
+    const pitch = sortedUniqueFontSizes.length === 1 ? 1 : 2 * sortedUniqueFontSizes.indexOf(thisFontSize) / (sortedUniqueFontSizes.length - 1);
+    return pitch.toFixed(2);
+};
+
+/** For each text element, set its pitch value in the "data-pitch" attribute
+ * @param {WordleInstance} instance - The Wordle instance that median, min and max font sizes are retrieved from.
+ * @param {Object[]} textElements - An array of DOM elements for wordle texts
+ */
+inverted_wordles.setPitches = function (instance, textElements) {
+    textElements.forEach(elm => {
+        elm.setAttribute("data-pitch", inverted_wordles.calculatePitch(inverted_wordles.getFontSizeValue(elm), instance.sortedUniqueFontSizes));
```

Comment:
Actually I think that the computation time of calculatePitch is negligible and we can do without storage in the "data-pitch" attribute, this will remove a bit of complexity